ipn/ipnlocal/peerapi.go

```diff
@@ -778,6 +778,9 @@ func (h *peerAPIHandler) canPutFile() bool {
 // canDebug reports whether h can debug this node (goroutines, metrics,
 // magicsock internal state, etc).
 func (h *peerAPIHandler) canDebug() bool {
+	if !slices.Contains(h.ps.selfNode.Capabilities, tailcfg.CapabilityDebug) {
```

Comment:
when does a new peerAPIServer get created? because ps.selfNode isn't updated in-place or on ps, so if I update ACL noteAttrs server-side, when does this take effect?

Reply:
Yeah, I think you want to do a call into LocalBackend instead to ask about the current netmap's self node capabilities, rather than trusting this h.ps.selfNode to be recent.

Reply:
Done, but we should remove ps.selfNode in that case. Will send follow up PR.